jax/random.py

```diff
@@ -367,6 +367,24 @@ def shuffle(key, x, axis=0):
   """
   return _shuffle(key, x, axis)
 
+
+def permutation(key, x):
+  """Permute the elements of an array along its first axis or return a permuted range
+
+  Args:n
+    key: a PRNGKey used as the random key.
+    x: the array or integer range to be shuffled.
+
+  Returns:
+    A shuffled version of x or array range
+  """
+  if isinstance(x, (int, onp.integer)):
+    arr = onp.arange(x)
+  else:
```

Comment:
Don't you want this shuffle to occur regardless of whether the array comes from an `arange` call?

Reply:
Absolutely, thanks for that. I forgot the fix the test, danger of copy-pasting.

Reply:
In the spirit of making this API slightly more explicit and less magically-polymorphic than NumPy's API, what if we just required the user to pass an array for `x`, i.e. require them to write `np.arange` when needed?

Reply:
Nevermind, I guess that's the whole point of this wrapper!